### PR TITLE
CET-4382 Scorecards details page

### PR DESCRIPTION
**Pull request by igorRoog**
## Change description

JIRA: CET-4382
Scorecard details page UI revamped.

## Screenshots
image
image
image
image
image
image
image
image
image

### CODE REVIEW

File: src/api/CortexClient.ts [hunk 1]
```diff
@@ -220,7 +221,7 @@ export class CortexClient implements CortexApi {
   }
 
   async getScorecard(scorecardId: number): Promise<Scorecard> {
-    return await this.get(`/api/backstage/v1/scorecards/${scorecardId}`);
+    return await this.get(`/api/backstage/v2/scorecards/${scorecardId}`);
```
[the remark this answers is not in the document]
Comment (igorRoog) [reply src/api/CortexClient.ts:224]: You're right! Initially, I wanted to extend v1, but we decided to create new v2. I'll change it!

[the remark this answers is not in the document]
Comment (igorRoog) [reply src/api/CortexClient.ts:224]: I don't get it. I've changed `Scorecard` type definition to match v2. I can see now, Scorecard is also returned by `getScorcards()`, so we should create new v2 of `getScorcards()`, to match new type.

File: src/components/Common/StatsItem.tsx [hunk 2]
```diff
@@ -60,7 +60,7 @@ const StatsItem: React.FC<StatsItemProps> = ({
   return (
     <Box>
       <CaptionTypography variant="caption">{caption}</CaptionTypography>
-      <Typography>
+      <Typography aria-label={caption}>
```
[the remark this answers is not in the document]
Comment (igorRoog) [reply src/components/Common/StatsItem.tsx:63]: Yes, right. That was my idea for testing content of this component. But I'll switch to data-testid. 👍

File: src/api/CortexClient.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (FoodProduct) [reply src/api/CortexClient.ts:224]: Oops sorry I noticed that after I wrote this comment and forgot to delete the comment. My mistake 🙈

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardMetadataCard/ScorecardMetadataFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+} from './ScorecardMetadataUtils';
+
+interface ScorecardMetadataFilterProps {
+  scorecard: Scorecard;
+}
+
+const useStyles = makeStyles(() => ({
+  ruleQuery: {
+    fontFamily:
+      'Soehne Mono, Input, Consolas, "Liberation Mono", Menlo, monospace',
```
[the remark this answers is not in the document]
Comment (igorRoog) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardMetadataCard/ScorecardMetadataFilter.tsx:35]: Would it be better to skip `Soehne` and just set it to `'Input, Consolas, "Liberation Mono", Menlo, monospace'`? Or do you have any other ideas?

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardMetadataCard/ScorecardMetadataUtils.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+export const getJoinerString = (idx: number, length: number) => {
+  return `${idx !== 0 && length > 2 ? ', ' : ''} ${
+    idx === length - 1 && length >= 2 ? 'and ' : ''
+  }`;
+};
+
+export const getEntityGroupsFromFilter = (
+  filter: EntityFilter | null,
+): [string[], string[]] => {
```
[the remark this answers is not in the document]
Comment (igorRoog) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardMetadataCard/ScorecardMetadataUtils.tsx:32]: Right, just copy-pasted from `brain-app`...

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardMetadataCard/ScorecardMetadataUtils.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+    return [entityGroups, excludeServiceGroups];
+  }
+
+  return [[], []];
+};
+
+export const getResourceTypesFromFilter = (
+  filter: EntityFilter | null,
+): [boolean, string[]] => {
```
[the remark this answers is not in the document]
Comment (igorRoog) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardMetadataCard/ScorecardMetadataUtils.tsx:51]: Same, copy-pasted from brain-app.

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardReportsTab/ScorecardReportsTab.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+export const ScorecardReportsTab = ({
+  scorecard,
+}: ScorecardReportsTabProps) => {
+  return (
+    <ItemCardGrid>
+      <ReportsPageCard
+        name="Bird's Eye"
+        description="Dive into your Scorecards to get insight into performance, broken down by teams, groups and rules and visualized as a heatmap."
```
[the remark this answers is not in the document]
Comment (igorRoog) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardReportsTab/ScorecardReportsTab.tsx]: @FoodProduct  https://github.com/cortexapps/brain-app/pull/4576

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardMetadataCard/ScorecardMetadataFilter.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (FoodProduct) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardMetadataCard/ScorecardMetadataFilter.tsx]: Is the goal here just to monospace? I'd see what other places in Backstage / Material Design do and try to copy that.

[the remark this answers is not in the document]
Comment (igorRoog) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardMetadataCard/ScorecardMetadataFilter.tsx]: It looks like `fontFamily: 'Monospace'` will do the job:
https://mui.com/system/typography/#font-family

**Review by FoodProduct — APPROVED** [review]
I didn't pull down to test, but the code looks great 👍. Thanks for taking on all this work! My understanding is that Cristina will be helping you test this more thoroughly soon but giving the approval here in case it all looks good 🚀

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardDetails.tsx [hunk 7]
```diff
@@ -56,32 +75,111 @@ export const ScorecardDetails = ({
...
+          />
+        </Tabs>
+        {activeTab === ScorecardDetailsTab.Scores && (
+          <Box display="flex" flexDirection="row" gridGap={8}>
+            <Button
+              onClick={() => setIsFilterDialogOpen(true)}
+              variant="outlined"
+              aria-label="Filter"
+            >
+              filter
```
[the remark this answers is not in the document]
Comment (igorRoog) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardDetails.tsx:170]: Not necessarily, as text inside MUI Button is capitalized.

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardDetails.tsx [hunk 8]
```diff
@@ -56,32 +75,111 @@ export const ScorecardDetails = ({
...
+          </Box>
         </Grid>
       </Grid>
+      <Box display="flex" justifyContent="space-between" alignItems="center">
+        <Tabs
+          value={activeTab}
+          indicatorColor="primary"
+          textColor="primary"
+          onChange={handleTabChange}
+          aria-label="Tabs for Scorecard"
```
[the remark this answers is not in the document]
Comment (igorRoog) [reply src/components/Scorecards/ScorecardDetailsPage/ScorecardDetails.tsx:144]: You're right, I'll remove it.

**Review by FoodProduct — APPROVED** [review]
Looks like we don't have the "Automatically dismiss stale reviews" setting enabled for this repo but re-approving anyway.

cc @cristinabuenahora is this something you can enable? I think we should, especially since this repo is public.

**Pull request merged**